### ISSUE TITLE
Consider allowing root usage

### ISSUE THREAD

**Issue by TheKevJames** [TheKevJames/puppet-homebrew#24]
From @jordigg :

I wouldn't let the user select user and/or group and instead move everything to be under the root user and the admin group: http://apple.stackexchange.com/questions/42127/homebrew-permissions-multiple-users-needing-to-brew-update

Something I do too is to set the default set up path to be `/Applications` rather than the `/Users/user/Applications` so everything that gets installed is available on the Applications folder of all users and is always located on the same applications folder `/Applications/`. I do that by setting an environment variable called `HOMEBREW_CASK_OPTS --appdir=/Applications`. Maybe we could allow the parameter `--appdir` when installing packages. Like `brew cask install --appdir="~/Applications" google-chrome`

**Comment by jordigg** [comment]
Here the changes to make it work with user `root` and multiple users that are part of the `admin` group.

https://github.com/improbable-io/puppet-homebrew/tree/allow-multiuser

Check and comment and I'll send a PR.

**Comment by TheKevJames** [comment]
I like the idea (and especially thanks for the `recurse => true`, I didn't realize that how it worked with directory targets!), but:

- I'm not sure how this makes the root user work properly? 
- I'm pretty sure the creates target on the new exec will cause problems: ie. a race condition between `sgid-brew-folders` and `install-homebrew`, since the former would never run if the later has been applied.
- Is the `sgid-brew-folders` exec even necessary given the changes to the file mode?

I grabbed the cleanup stuff from your branch and put it into #31.

**Comment by jordigg** [comment]
I have my version running on some mac minis acting as Teamcity build agents and they work like a charm. No more complains when I run `brew doctor`.

I'll try to paste a link here with the best practice but I think I remember that the recommended permissions on /usr/bin are `root` as owner and `staff` or `admin` as group. I recommend using the `admin` group since all administrators on the mac should be able to install and manage applications via brew.

I think you are right. We can remove the
```
  exec { 'sgid-brew-folders':
    command   => '/bin/chmod g+w /usr/local && /bin/chmod g+w /Library/Caches/Homebrew',
    creates   => '/usr/local/bin/brew',
    logoutput => on_failure,
    timeout   => 0,
    require   => [ File['/usr/local'], File['/Library/Caches/Homebrew'] ],
  }
```
since `chmod 755` already takes care of the correct ownership and permissions for owner and group. I'll test and report back.

**Comment by jordigg** [comment]
Tested and works for me. Here the final version of install.pp
https://github.com/TheKevJames/puppet-homebrew/blob/ce3d681ba300011eab58058e86caeba220f9d980/manifests/install.pp

**Comment by TheKevJames** [comment]
Cool! Likes like the only change on top of #31 would be the `chmod 0755`. I'm going to hold off on making `root` the default user (I think homebrew's recommendation of avoiding using the root user should be mirrored here), but I'll update the README to talk about this.

**Comment by jordigg** [comment]
That's fine with me. Let the user choose and don't specify any on the init like you have it now. Same with group but I'm confident we can leave 'admin' as the sefault one.
Throw error if the user isn't specified.

**Comment by TheKevJames** [comment]
Released in 1.4.1. Please let me know if anything behaves unexpectedly.

**Comment by jordigg** [comment]
Perfect! I'll test tomorrow and report if I see anything strange. Thanks!
